src/main/java/org/jenkinsci/plugins/authorizeproject/AuthorizeProjectProperty.java

```diff
@@ -119,15 +135,60 @@ public Authentication authenticate(Queue.Item item) {
         }
         return strategy.authenticate(owner, item);
     }
-    
-    @Initializer(after=InitMilestone.PLUGINS_STARTED)
+
+    /**
+     * Ensure that deserialization failures in this field result in a failure to deserialize the job.
+     * This method is responsible for ensuring that POSTing config.xml respects the defined strategy.
+     */
+    @Initializer(after = InitMilestone.PLUGINS_STARTED)
     public static void setStrategyCritical() {
         Items.XSTREAM2.addCriticalField(AuthorizeProjectProperty.class, "strategy");
     }
-    
+
+    /**
+     * If we are being deserialized outside of loading the initial jobs (or reloading) then we need to cross check
+     * the strategy permissions to defend against somebody trying to push a configuration relating to a user other
+     * than themselves.
+     *
+     * @return {@code this}
+     * @throws ObjectStreamException if the object cannot be deserialized.
+     * @see #setStrategyCritical()
+     */
+    private Object readResolve() throws ObjectStreamException {
+        if (strategy != null) {
+            Authentication authentication = Jenkins.getAuthentication();
+            if (authentication != ACL.SYSTEM) {
+                if (ProjectQueueItemAuthenticator.isConfigured()) {
+                    StaplerRequest request = Stapler.getCurrentRequest();
+                    AccessControlled context;
+                    if (request == null) {
+                        context = Jenkins.getActiveInstance();
+                    } else {
+                        Job job = request.findAncestorObject(Job.class);
+                        context = job == null ? Jenkins.getActiveInstance() : job;

```

Comment:
`job` gets `null` when configured via CLI.
It can results a problem in the following case:
* User A don't have Job.BUILD permission in the system global scope.
* User A have Job.BUILD permission for project X (using ProjectMatrixAuthorizationStrategy)
* project X is configured to use "Run as user who triggered the build"
* User A tries to update the configuration of project X with CLI -> Fails!

Though, There's no way for that, and it should be accepted as limitation.